src/awe/rmdparchive.h

```diff
@@ -164,6 +173,42 @@ class RMDPArchive : public Archive {
 	std::vector<FolderEntry> _folderEntries;
 	std::vector<FileEntry> _fileEntries;
 
+	/*!
+	 * A helper function that navigates through _folderEntries under
+	 * given path.
+	 *
+	 * \return folder entry, if it exists under giver path
+	 */
+	std::optional<FolderEntry> findDirectory(std::string &path) const;
+
+	/*!
+	 * A helper function that fills in known FolderEntry fields.
+	 * 
+	 * \param readUint32 A ReadStream member function to read
+	 * 32-bit unsigned integers with appropriate endianness
+	 * \param nameSize Max name size in bytes, obtained from
+	 * earlier file headers.
+	 */
+	FolderEntry readFolder(Common::ReadStream *bin, Common::EndianReadStream end, uint32_t nameSize);
+
+	/*!
+	 * A helper function that fills in known FileEntry fields.
+	 * 
+	 * \param readUint32 A ReadStream member function to read
+	 * 32-bit unsigned integers with appropriate endianness
+	 * \param readUint64 A ReadStream member function to read
+	 * 64-bit unsigned integers with appropriate endianness
+	 * \param nameSize Max name size in bytes, obtained from
+	 * earlier file headers.
+	 */
+	FileEntry readFile(Common::ReadStream *bin, Common::EndianReadStream end, uint32_t nameSize);
+
+	/*!
+	 * A helper functon that find a file in a given folder by
+	 * its name hash value.
+	 */
+	std::optional<FileEntry> findFile(const FolderEntry &folder, const uint32_t nameHash) const;
```

Comment:
I will move them higher, but I have to put them at least below the structs. Since those functions either pass or return `FolderEntry` or `FileEntry`, putting them above said structs leads to a compilation error.